### PR TITLE
Fixed #225 (Phone screen background)

### LINKED ISSUE #225: Adding a "A background display" to the "Change size of preview feature"
**Description**
The preview feature is great, and works like a charm. I do think, however, there is a tiny scope of improvement! I've worked hard in the past on making websites responsive, and I believe it helps reach a bigger audience fairly quickly. 

I would suggest adding a phone-esque background to the preview mode. Something like this:-
 
load

**Problems solved**
From a developers perspective, It is important to be able to judge dimensions and make things look good on all devices. And what better way to be able to visualise it when you can have a mock phone and screen to be able to look at? 

**Alternatives**
If getting these backgrounds proves to be difficult, we could start off with a dummy phone background, which applies to all the cases, and slowly diversifies.

**Additional context**
This is how it looks as of now:-

send

**Want to take up?**
Yes.

### PR DESCRIPTION
**Pull request by nimishbongale**
## Fixes
Fixes #225 by @nimishbongale

## Description
I've added in phone screen backgrounds to all the available devices (Phones+Tablets+Laptops) from  the devices menu. 

## Technical details
This was achieved using CSS. The viewports support inbuilt styling for these purposes. 
SVG's was a great option too, but It could have involved bloating the repo (as was previously mentioned). Also, at the moment, `storybook.js` does not provide for the full customisation of the canvas which is used as the background for the viewports. There's full control over the styles of the viewport, and I have utilised this property in order to get the needful done.   

The dimensions actually correspond to some (slightly scaled down) versions of the real world entitites. I have taken my measurements from the official company websites and some from:

* https://www.dimensions.guide/

## Tests
Running storybook on the fork should be sufficient. The CD setup also works.

## Screenshots

Here's a screenshot for IPhone 5 and Samsung Galaxy 9 :

iphone

### IPhone 5

galaxy

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .storybook/viewport.js [hunk 1]
```diff
@@ -7,62 +7,91 @@ export default {
     styles: {
       width: '320px',
       height: '480px'
-    },
-    type: 'mobile'
+      },
+      type: 'mobile'
```
Comment (dhruvkb) [inline .storybook/viewport.js:11]: Not a real change.

File: .storybook/viewport.js [hunk 2]
```diff
@@ -7,62 +7,91 @@ export default {
...
   },
-  iPhoneSE: {
+  iPhoneSE: {
```
Comment (dhruvkb) [inline .storybook/viewport.js:13]: Remove trailing spaces.

File: .storybook/viewport.js [hunk 3]
```diff
@@ -7,62 +7,91 @@ export default {
...
     },
     type: 'mobile'
   },
   iPhone11Pro: {
     name: 'Apple iPhone 11 Pro',
     styles: {
-      width: '375px',
-      height: '812px'
+      width: '414px',
+      height: '896px',
```
Comment (dhruvkb) [inline .storybook/viewport.js:56]: Can you cite sources for this change?

File: .storybook/viewport.js [hunk 4]
```diff
@@ -72,47 +101,69 @@ export default {
     name: 'Classic tablet',
     styles: {
       width: '600px',
-      height: '800px'
+      height: '800px',
```
Comment (dhruvkb) [inline .storybook/viewport.js:104]: Remove trailing comma

File: .storybook/viewport.js [hunk 5]
```diff
@@ -127,74 +178,113 @@ export default {
     type: 'monitor'
   },
   AsusChromebook: {
-    name: 'Asus Chrombook',
+    name: 'Asus Chromebook',
```
Comment (dhruvkb) [inline .storybook/viewport.js:181]: Ah! Good catch!

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Can you add the bezels for the other devices and screens as well?

File: .storybook/viewport.js (hunk 1 above)
Comment (nimishbongale) [reply .storybook/viewport.js:11]: My bad, didn't intend on making that change.

File: .storybook/viewport.js (hunk 2 above)
Comment (nimishbongale) [reply .storybook/viewport.js:13]: Alright, will do.

File: .storybook/viewport.js (hunk 3 above)
Comment (nimishbongale) [reply .storybook/viewport.js:56]: The IPhone 11 had the same dimensions as compared  to the IPhone 11 Pro before I made these changes. In reality, IPhone 11 Pro is just slightly smaller than IPhone 11. (5.8 v/s 6.1). Will make the necessary changes!

File: .storybook/viewport.js (hunk 4 above)
Comment (nimishbongale) [reply .storybook/viewport.js:104]: WIll do.

File: .storybook/viewport.js (hunk 3 above)
Comment (dhruvkb) [reply .storybook/viewport.js:56]: Yeah, screen sizes are generally independent of screen resolutions. Changing the resolution to mimic physical dimensions is not correct.

Comment (nimishbongale) [reply .storybook/viewport.js:56]: Agreed. Will revert back this change.

File: .storybook/viewport.js (hunk 1 above)
Comment (nimishbongale) [reply .storybook/viewport.js]: @dhruvkb Done!

File: .storybook/viewport.js (hunk 2 above)
Comment (nimishbongale) [reply .storybook/viewport.js]: Done!

File: .storybook/viewport.js (hunk 3 above)
Comment (nimishbongale) [reply .storybook/viewport.js]: @dhruvkb Reverted. Fixed!

File: .storybook/viewport.js (hunk 4 above)
Comment (nimishbongale) [reply .storybook/viewport.js]: Done.

**Review by dhruvkb — APPROVED** [review]
This looks about right, and the solution using CSS borders is much better than SVG, but I'm still a little unsure about how you arrived at these pixel values.

**Comment by dhruvkb** [comment]
@nimishbongale can you give me a little clarity behind your border calculations so I can merge? Thanks and sorry for dragging this out.

**Comment by nimishbongale** [comment]
@dhruvkb will do it by tonight. My shabby calculations on a sheet of paper aren't helpful (by any means) to help someone understand what I've done and how I've done it. Will put up the explanation here asap!

**Comment by nimishbongale** [comment]
> @nimishbongale can you give me a little clarity behind your border calculations so I can merge? Thanks and sorry for dragging this out.

Its completely justified on your part to be asking for more clarifications as you people are the repository maintainers and all the code in the repository and its functionality must be known to you. 

Here's the explanation:

I'd like to take an example of Galaxy Note 9. Rest assured, the remaining calculations for the other phones/tablets/PC's are done in a similar manner. 

* Width of Galaxy Note 9 = **76.4mm** 
* Height of Galaxy Note 9 = **161.9mm**
* Screen width in px = **1440px** (CSS Width is **720px**, scaled down by a native factor of 2)
* Screen height in px = **2960px** (CSS Height is **1480px**, scaled down by a native factor of 2, again)
* Screen resolution in ppi = **516ppi**
 
Using the screen width, height and resolution, (or the 6" diagonal) you can figure out the dimensions of the mobile screen, which is about **71mm x 146mm**. It's simple math, but I used 
[pixelcalculator](http://pixelcalculator.com/) just to be certain.

This means that you are left with **5.4mm** (76.4-71) width bezels on the left and right edges and about **15.9mm** (161.9-146) width bezels on the top and bottom edges. Now the only thing left to do is convert these back to pixels to feed into our CSS. Here's where the tricky part sets in. 

Just a side note, ppi/dpi/xdpi are often confused with each other because the terms are so close to each other and are often used interchangeably. The major thing that differentiates ppi from dpi is the context in which it is used. ppi (or pixels per inch squared) is used in the digital formats, while dpi (dots per inch squared) is commonly used to describe the resolution of printed copies. 

96ppi is generally considered a standard for web or web designs. As times changed, devices with better and better resolutions kept coming up. Manufacturers looked to cramp in as many pixels as possible in the limited space they had. This encouraged a rather generous shift in the thought processes of manufacturers towards building better graphics, displays and resolutions.

For the sake of standardisation, I used 96ppi to convert the mm back to pixel values. What this means is that any device (and subsequenty any brower) having a resolution of 96ppi+ will be able to view the viewport in the canvas just fine (without it overshooting the canvas boundaries). This is again, scaling maths, but I used [pixelsconverter](https://pixelsconverter.com/millimeters-to-pixels) just to be certain. 

Feeding 5.4mm into the converter at 96ppi gives me a value of **20px** which means about **10px** on either side. It looked a bit boxed up to me to be fair, so I ended up reducing it to **7.5px** on either side, but this also accounts for the curved edges Note 9 has.    

Feeding 15.9mm into the converter at 96ppi gives me a value of **60px** which means about **30px** on either side. The Note 9 does not have an even split between the top and bottom bezels, so I changed it to **35px** on top, and **25px** on the bottom. 

As far as the border radius goes, that was obtained by eyeballing it. The human eye is great in noticing things that lack curvature/are not curved enough. I also happened to share some screenshots with my friends about the design and asked them to identify the phone (from just the bezels). They were actually able to guess them!

I hope this satisfies your query regarding the dimensions. I have gone to the extent of using the netlify build copy [deploy-preview](https://deploy-preview-227--cc-vue-vocabulary.netlify.app/) and used the chrome dev tools to check if it looks fine on all the devices mentioned.

Thanks!

**Comment by dhruvkb** [comment]
That right there is some heavy duty mathematics! While most of it went right over my head in a casual read, it seems you know your stuff and the devices look fairly accurate to me as well.

**Comment by nimishbongale** [comment]
> That right there is some heavy duty mathematics! While most of it went right over my head in a casual read, it seems you know your stuff and the devices look fairly accurate to me as well.

@dhruvkb 

Well, I'm glad I was able to get it done as well! It did take some considerable time and effort, especially when I chose the CSS path over the SVG. SVG's would've ended up bloating the repository, for sure. 

I'm really happy you asked me to delve deep into the math/justify my calculations. Quite a contrary to the general notion of "If it looks right, It is right!". Happy to be on the contributors list! Thanks a ton for appreciating the contrbution!

Thank you for all your help! :) 

Looking forward on configuring dependabot as well. Let me know if it can be done by me, or it requires someone with "direct write access" to the repository. Surely I can setup the `.github/dependabot.yml`, and I think you might have the enable dependabot manually? Will look into this.

**Comment by dhruvkb** [comment]
Yeah, @nimishbongale please open an issue and replicate this the CC Vocabulary repo as well. We generally try to keep features at parity in both repos.

**Comment by nimishbongale** [comment]
@dhruvkb Yes. On it! Will be done in a bit.

File: .storybook/viewport.js [hunk 6]
```diff
@@ -80,39 +109,61 @@ export default {
...
     styles: {
       width: '810px',
-      height: '1080px'
+      height: '1080px',
+      borderTopWidth: '85px',
+      borderBottomWidth: '85px',
+      borderLeftWidth: '20px',
+      borderRightWidth: '20px',
+      borderRadius: '45px'
+
```
Comment (dhruvkb) [inline .storybook/viewport.js:131]: Should have caught this before but please remove this blank line here.

**Review by dhruvkb — COMMENTED** [review]
Post-merge changes:

File: .storybook/viewport.js (hunk 6 above)
Comment (nimishbongale) [reply .storybook/viewport.js:131]: Alright!

WIll do. Slipped through my eyes as well.

**Pull request merged**